### PR TITLE
[INLONG-8239][Sort] Support sort format to flink-1.15

### LINKED ISSUE #8239: [Improve][Sort] Support sort format to flink-1.15
### Description

_No response_

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by GanfengTan**
Support sort format flink 1.13 and 1.15, the default package is flink 1.13.
 If packaging flink1.15 please use -P v1.15 

- Fixes #8239 

### Motivation

Support multiple versions of flink.

### Modifications

Add new module and modify the POM.

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

### CODE REVIEW

File: inlong-sort/sort-dist/pom.xml [hunk 1]
```diff
@@ -145,4 +164,172 @@
...
+                </dependency>
+                <dependency>
+                    <groupId>org.apache.flink</groupId>
+                    <artifactId>flink-sql-orc_${scala.binary.version}</artifactId>
+                    <version>1.13.5</version>
+                </dependency>
+            </dependencies>
+        </profile>
+        <profile>
+            <id>v1.5</id>
```
Comment (gong) [inline inlong-sort/sort-dist/pom.xml:251]: v1.5 -> v1.15

File: inlong-sort/sort-formats/format-json-v1.5/pom.xml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.inlong</groupId>
+        <artifactId>sort-formats</artifactId>
+        <version>1.8.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>sort-format-json-v1.5</artifactId>
+    <name>Apache InLong - Sort Format-json-v1.5</name>
```
Comment (gong) [inline inlong-sort/sort-formats/format-json-v1.5/pom.xml:30]: Maybe v1.5 -> v1.15

File: inlong-sort/sort-dist/pom.xml [hunk 3]
```diff
@@ -80,6 +82,11 @@
             <artifactId>sort-format-json</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.inlong</groupId>
+            <artifactId>sort-format-json-v1.5</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
Comment (gong) [inline inlong-sort/sort-dist/pom.xml:89]: ` sort-format-json` and `sort-format-json-v1.5`will cause conflict with v1.13 when package -pv1.1.3.

File: inlong-sort/sort-dist/pom.xml (hunk 1 above)
Comment (GanfengTan) [reply inlong-sort/sort-dist/pom.xml]: 3qs

File: inlong-sort/sort-formats/format-json-v1.5/pom.xml (hunk 2 above)
Comment (GanfengTan) [reply inlong-sort/sort-formats/format-json-v1.5/pom.xml]: yes

File: inlong-sort/sort-dist/pom.xml (hunk 3 above)
Comment (GanfengTan) [reply inlong-sort/sort-dist/pom.xml]: Profile will distinguish between packaging.

File: inlong-sort/sort-dist/pom.xml [hunk 4]
```diff
@@ -248,86 +159,25 @@
...
-                </dependency>
-                <dependency>
-                    <groupId>org.apache.inlong</groupId>
-                    <artifactId>sort-format-inlongmsg-csv</artifactId>
-                    <version>${project.version}</version>
-                </dependency>
                 <dependency>
                     <groupId>org.apache.inlong</groupId>
                     <artifactId>sort-format-json-v1.5</artifactId>
                     <version>${project.version}</version>
```
Comment (gong) [inline inlong-sort/sort-dist/pom.xml]: v1.5 -> v1.15

File: inlong-sort/sort-dist/pom.xml [hunk 5]
```diff
@@ -145,4 +133,53 @@
...
+            <id>v1.15</id>
+            <properties>
+                <flink.version>1.15.4</flink.version>
+            </properties>
+            <dependencies>
+                <dependency>
+                    <groupId>org.apache.inlong</groupId>
+                    <artifactId>sort-format-json-v1.5</artifactId>
+                    <version>${project.version}</version>
+                </dependency>
```
Comment (gong) [inline inlong-sort/sort-dist/pom.xml:171]: v1.5 -> v1.15

File: inlong-sort/sort-formats/format-inlongmsg-base/src/main/java/org/apache/inlong/sort/formats/inlongmsg/InLongMsgValidator.java [hunk 6]
```diff
@@ -21,17 +21,20 @@
...
  * Validator for mixed inlongmsg formats.
  */
-public class InLongMsgValidator extends FormatDescriptorValidator {
+public class InLongMsgValidator implements DescriptorValidator {
 
     @Override
     public void validate(DescriptorProperties properties) {
-        super.validate(properties);
-
+        properties.validateString(FORMAT_TYPE, false, 1);
```
Comment (EMsnap) [inline inlong-sort/sort-formats/format-inlongmsg-base/src/main/java/org/apache/inlong/sort/formats/inlongmsg/InLongMsgValidator.java:36]: duplicate code with InLongMsgMixedValidator.java

**Comment by EMsnap** [comment]
Maybe if we use format-json-v1.15 to identify flink1.15, then also we should use format-json-v1.13 to identify flink1.13

File: inlong-sort/sort-dist/pom.xml (hunk 5 above)
Comment (GanfengTan) [reply inlong-sort/sort-dist/pom.xml]: done

File: inlong-sort/sort-formats/format-inlongmsg-base/src/main/java/org/apache/inlong/sort/formats/inlongmsg/InLongMsgValidator.java (hunk 6 above)
Comment (GanfengTan) [reply inlong-sort/sort-formats/format-inlongmsg-base/src/main/java/org/apache/inlong/sort/formats/inlongmsg/InLongMsgValidator.java:36]: Sorry, I don't understand what you mean.

Comment (EMsnap) [reply inlong-sort/sort-formats/format-inlongmsg-base/src/main/java/org/apache/inlong/sort/formats/inlongmsg/InLongMsgValidator.java:36]: InLongMsgMixedValidator have duplicate code with InlongMsgValidator could you please extract these code ?

**Review by healchow — APPROVED** [review]

File: inlong-sort/sort-formats/format-inlongmsg-base/src/main/java/org/apache/inlong/sort/formats/inlongmsg/InLongMsgValidator.java (hunk 6 above)
Comment (GanfengTan) [reply inlong-sort/sort-formats/format-inlongmsg-base/src/main/java/org/apache/inlong/sort/formats/inlongmsg/InLongMsgValidator.java]: done

**Review by gong — APPROVED** [review]

**Comment by GanfengTan** [comment]
> Maybe if we use format-json-v1.15 to identify flink1.15, then also we should use format-json-v1.13 to identify flink1.13

A PR will be submitted again for overall version differentiation.

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Comment by GanfengTan** [comment]
> > Maybe if we use format-json-v1.15 to identify flink1.15, then also we should use format-json-v1.13 to identify flink1.13
> 
> A PR will be submitted again for overall version differentiation.

Thank you for your suggestion, there will be a PR to distinguish and extract versions.

File: inlong-sort/sort-dist/pom.xml [hunk 7]
```diff
@@ -31,6 +31,8 @@
 
     <properties>
         <inlong.root.dir>${project.parent.parent.basedir}</inlong.root.dir>
+        <flink.v15.version>1.15.4</flink.v15.version>
```
[the remark this answers is not in the document]
Comment (GanfengTan) [reply inlong-sort/sort-dist/pom.xml:34]: https://github.com/apache/inlong/pull/8246#issuecomment-1605850333
Thank you for your suggestion, there will be a PR to distinguish and extract versions.

**Pull request merged**